site/src/main/java/uk/nhs/digital/ps/beans/BaseDocument.java

```diff
@@ -39,4 +44,22 @@ public HippoBean getSelfLinkBean() {
     protected void assertPropertyPermitted(String propertyName) {
         // To be overwritten by subclasses for specific implementation
     }
+
+    /**
+     * Converts given {@linkplain Calendar} to {@linkplain RestrictableDate}.
+     */
+    protected RestrictableDate nominalPublicationDateCalendarToRestrictedDate(final Calendar calendar) {
```

Comment:
Hmm, we seem to be adding more methods to `BaseDocument` that not all its extending classes use - this applies to this and all the `assertPropertypermitted`-related methods. For example, `Series` doesn't use any of them and yet it inherits - and exposes - them.

I wonder if this is the time where we should consider applying composition over inheritance and move this and `assertPropertypermitted`-related methods to an external class, where `Publication` and `Dataset` would call/wrap them rather than inherit them from the base class. While the `assertPropertypermitted` & Co. are arguably more generic in nature, the `nominalPublicationDateCalendarToRestrictedDate` is much more specific.

I worry that if we continue this trend, at some point we will make BaseDocument so bloated, and it's responsibilities so blurred, that we'll find it hard to understand and maintain. 

BTW, it's my fault that we've started down this path as, if I remember correctly, it was me who advised Jake to stick the `assertPropertypermitted`-related methods here :wink:

Happy to talk about it more if you want and if you think there is enough merit in the above comment.

Reply:
something to consider for next PR :D